pkg/util/proto/testing/swagger_oneof.json

```diff
@@ -0,0 +1,49 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Kubernetes",
+    "version": "v1.8.0"
+  },
+  "paths": {},
+  "definitions": {
+    "A": {
+      "description": "The first letter of the alphabet.",
+      "type": "object",
+      "properties": {
+        "apple": {
+          "description": "The round fruit of a tree of the rose family, which typically has thin red or green skin and crisp flesh. Many varieties have been developed as dessert or cooking fruit or for making cider.",
+          "type": "string"
+        },
+        "apricot": {
+          "description": "A juicy, soft fruit, resembling a small peach, of an orange-yellow color.",
+          "type": "string"
+        },
+        "fruit": {
+          "description": "The sweet and fleshy product of a tree or other plant that contains seed and can be eaten as food.",
+          "type": "string"
```

Comment:
I think the discriminator marker should maybe go here. This field is super confusing. I spent longer than I'd care to say trying to figure out why apple, apricot, and fruit were in the same list.

Reply:
Yeah, the more I think about it, the more I disagree (sorry :-)). The discriminator is really not a proprety of the field, but a property of the struct that holds it.

The struct is a oneof that will discriminate on field Y. OpenAPI v3 has a "discriminator" field that gives the property, and it's at the "oneof" level.